core/src/wallet/stellar/transaction_builders/StellarLikeTransactionBuilder.cpp

```diff
@@ -0,0 +1,179 @@
+/*
+ *
+ * StellarLikeTransactionBuilder.cpp
+ * ledger-core
+ *
+ * Created by Pierre Pollastri on 27/08/2019.
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include "StellarLikeTransactionBuilder.hpp"
+#include <wallet/stellar/StellarLikeAccount.hpp>
+#include <api/StellarLikeTransactionCallback.hpp>
+#include <algorithm>
+#include <api_impl/BigIntImpl.hpp>
+
+namespace ledger {
+    namespace core {
+
+        StellarLikeTransactionBuilder::StellarLikeTransactionBuilder(
+                const std::shared_ptr<StellarLikeAccount> &account) : _account(account) {
+            // Initialize everything with default values
+            _envelope.tx.memo.type = stellar::xdr::MemoType::MEMO_NONE;
+        }
+
+        std::shared_ptr<api::StellarLikeTransactionBuilder>
+        StellarLikeTransactionBuilder::putSignature(const std::vector<uint8_t> &signature) {
+            stellar::xdr::DecoratedSignature sig;
+            sig.signature = signature;
+            _envelope.signatures.emplace_back(sig);
+            return shared_from_this();
+        }
+
+        std::shared_ptr<api::StellarLikeTransactionBuilder>
+        StellarLikeTransactionBuilder::addNativePayment(const std::string &address,
+                                                        const std::shared_ptr<api::Amount> &amount) {
+            stellar::xdr::Operation operation;
+            stellar::xdr::PaymentOp op;
+            op.destination.type = stellar::xdr::PublicKeyType::PUBLIC_KEY_TYPE_ED25519;
+            StellarLikeAddress addr(address, _account->getWallet()->getCurrency(), Option<std::string>::NONE);
+            auto pubKey = addr.toPublicKey();
+            std::copy(pubKey.begin(), pubKey.begin() + op.destination.content.max_size(), op.destination.content.begin());
```

Comment:
potential `iterator overflow` here
can you check if `op.destination.content.max_size()` is strictly inferior to `pubKey.size()` ?

Reply:
One more question, `op.destination.content` is already allocated ? else please use `std::back_inserter`